ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/helpers/OffboardingHelperTest.java

```diff
@@ -0,0 +1,335 @@
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.support.membermodification.MemberMatcher.method;
+import static org.powermock.reflect.Whitebox.invokeMethod;
+import static org.powermock.reflect.Whitebox.setInternalState;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        OffboardingHelper.class,
+        Kayako.class
+})
+public class OffboardingHelperTest {
+    private OffboardingHelper underTest;
+    @Rule
+    private ExpectedException expectedException = ExpectedException.none();
+    @Mock
+    private Rating rating;
+    @Mock
+    private Resources resources;
+    @Mock
+    private Context context;
+    @Mock
+    private OffboardingHelper.OffboardingHelperViewCallback offboardingHelperViewCallback;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        staticMock();
+        underTest = spy(new OffboardingHelper());
+    }
+
+    @Test
+    public void onUpdateRating() throws Exception {
+        // Arrange
+        when(rating.getId()).thenReturn(42L);
+        when(rating.getScore()).thenReturn(Rating.SCORE.GOOD);
+        when(rating.getComment()).thenReturn("good");
+
+        doNothing().when(underTest, method(OffboardingHelper.class, "removeFromQueue"))
+                .withArguments(offboardingHelperViewCallback, Rating.SCORE.GOOD, "good");
+        // Act
+        underTest.onUpdateRating(rating, offboardingHelperViewCallback);
+        // Assert
+        ArgumentCaptor<Boolean> argumentCaptor = ArgumentCaptor.forClass(Boolean.class);
+        verify(offboardingHelperViewCallback).onRefreshListView(argumentCaptor.capture());
+        assertTrue(argumentCaptor.getValue());
+    }
+
+    @Test
+    public void onFailureToUpdateRating() {
+        // Act
+        underTest.onFailureToUpdateRating(Rating.SCORE.GOOD, "feedback",
+                offboardingHelperViewCallback);
+        // Assert cannot be done due to private calls
```

Comment:
🚫  This test doesn't contain a mechanism to assert the expected results or exceptional cases. This kind of tests creates artificial code coverage. To consider a test as valid, it must be prone to fail and it must have either assertions or expected exceptions. Consider adding assertions or expect specific exceptions in order to follow testing best practices.  

**[More details](https://confluence.devfactory.com/x/AxqkEw)**
          
## The solution in this case
Adding expectations on `offboardingHelperViewCallback` mock

*Proof of concept*
```JAVA
// Assert
verify(offboardingHelperViewCallback).onRefreshListView(true);

```
 

<a id="9398ebf0-5328-11e8-a476-4b3e34747147" href="none" codereview-uuid-reserved-tag />